matchzoo/embedding.py

```diff
@@ -1,143 +1,149 @@
-
-            # Go through all embedding file
-            for line in tqdm(embedding_file_ptr):
-                # Explicitly splitting on " " is important, so we don't
-                # get rid of Unicode non-breaking spaces in the vectors.
-                entries = line.rstrip().split(b" ")
-
-                word, vec = entries[0], entries[1:]
-
-                if self._embedding_dim != len(vec):
-                    raise RuntimeError(
-                        "Vector for token {} has {} dimensions, but "
-                        "previously read vectors have {} dimensions. "
-                        "All vectors must have the same number of "
-                        "dimensions.".format(word, len(vec),
-                                             self._embedding_dim))
-
-                try:
-                    if isinstance(word, bytes):
-                        word = word.decode('utf-8')
-                except Exception:
-                    logger.warning(
-                        "Skipping non-UTF8 token {}".format(repr(word)))
-                    continue
-
-                index = term_index.get(word, None)
-                if index:
-                    self._embedding_mat[index] = np.array(vec).astype(float)
-                    self._index_state[index] = 1
-                    num_shared_words += 1
+    def matrix(self):
+        """"""
+        return self._matrix
+    
+    @matrix.setter
+    def matrix(self, matrix_instance):
+        self._matrix = matrix_instance
+
+class RandomInitializedEmbedding(Embedding):
+    """
+    Random Initialized WordEmbedding.
+    """
+    def __init__(
+        self,
+        vocab: dict,
+        dimension: int,
+        random_scale: float = 0.2):
+        """Initialization."""
+        self._vocab = vocab # term index.
+        self._dimension = dimension
+        self._matrix = np.random.uniform(-random_scale,
+                                         random_scale,
+                                         (len(self._vocab), self._dimension))
+    
+    def _entity_to_vec(self, entity):
+        """"""
+        return self._matrix[self._vocab[entity]]
+
+class PretrainedEmbedding(Embedding, abc.ABC):
+    """Pretrained."""
+
+    def __init__(self):
+        """"""
+        self._vocab = self._matrix.index.values
+        self._dimension = self._matrix.shape[1]
+
+    def _entity_to_vec(self, entity):
+        """"""
+        return self._matrix.loc[entity].values
+    
+    @abc.abstractmethod
+    def _read_embedding(self, file_path: str):
+        """Read embedding file given dir."""
+
+
+class Word2Vec(PretrainedEmbedding):
+    """Word2Vec.
 
-            # init tht OOV word embeddings
-            for word in term_index:
-                index = term_index[word]
-                if index not in self._index_state:
-                    self._index_state[index] = 2
+    Examples:
+        >>> word2vec = Word2Vec('tests/sample/embed_10.txt')
```

Comment:
I moved `tests/sample` things on `2.0-dev`. You may keep this and let me change it later, or do it yourself.

Reply:
got it